invenio_vocabularies/services/config.py

```diff
@@ -0,0 +1,205 @@
+
+from flask import current_app
+from invenio_i18n import lazy_gettext as _
+from invenio_records_resources.services import (
+    Link,
+    LinksTemplate,
+    RecordService,
+    RecordServiceConfig,
+    SearchOptions,
+    pagination_links,
+)
+from invenio_records_resources.services.base import (
+    ConditionalLink,
+    Service,
+    ServiceListResult,
+)
+from invenio_records_resources.services.records.components import DataComponent
+from invenio_records_resources.services.records.params import (
+    FilterParam,
+    SuggestQueryParser,
+)
+from sqlalchemy import asc, desc
+
+from ..records.api import Vocabulary
+from . import results
+from .components import PIDComponent, VocabularyTypeComponent
+from .permissions import PermissionPolicy
+from .schema import TaskSchema, VocabularySchema
+
+
+def is_custom_vocabulary_type(vocabulary_type, context):
+    """Check if the vocabulary type is a custom vocabulary type."""
+    return vocabulary_type["id"] in current_app.config.get(
+        "VOCABULARIES_CUSTOM_VOCABULARY_TYPES", []
+    )
+
+
+class VocabularySearchOptions(SearchOptions):
+    """Search options for vocabularies."""
+
+    params_interpreters_cls = [
+        FilterParam.factory(param="tags", field="tags"),
+    ] + SearchOptions.params_interpreters_cls
+
+    suggest_parser_cls = SuggestQueryParser.factory(
+        fields=[
+            "id.text^100",
+            "id.text._2gram",
+            "id.text._3gram",
+            "title.en^5",
+            "title.en._2gram",
+            "title.en._3gram",
+        ],
+    )
+
+    sort_default = "bestmatch"
+
+    sort_default_no_query = "title"
+
+    sort_options = {
+        "bestmatch": dict(
+            title=_("Best match"),
+            fields=["_score"],  # ES defaults to desc on `_score` field
+        ),
+        "title": dict(
+            title=_("Title"),
+            fields=["title_sort"],
+        ),
+        "newest": dict(
+            title=_("Newest"),
+            fields=["-created"],
+        ),
+        "oldest": dict(
+            title=_("Oldest"),
+            fields=["created"],
+        ),
+    }
+
+
+class VocabularyTypeSearchOptions(SearchOptions):
+    """Search options for vocabulary types."""
+
+    # TODO: Is this still necessary here?
+    params_interpreters_cls = [
+        FilterParam.factory(param="tags", field="tags"),
+    ] + SearchOptions.params_interpreters_cls
+
+    # TODO: Is this still necessary here?
+    suggest_parser_cls = SuggestQueryParser.factory(
+        fields=[
+            "id.text^100",
+            "id.text._2gram",
+            "id.text._3gram",
+        ],
+    )
+
+    sort_options = {
+        "id": dict(
+            title=_("ID"),
+            fields=["id"],
+        ),
+    }
+
+    sort_default = "id"
+
+    sort_default_no_query = "id"
+
+    sort_direction_options = {
+        "asc": dict(title=_("Ascending"), fn=asc),
+        "desc": dict(title=_("Descending"), fn=desc),
```

Comment:
```suggestion
        "asc": dict(title=_("Ascending"), fn=sa.asc),
        "desc": dict(title=_("Descending"), fn=sa.desc),
```

nit: to keep the namespace cleaner (what I have in mind is `import sqlalchemy as sa`)